### PR TITLE
Lots of Fun-O-Meter dashboard improvements, including the creation of script and script_level views.

### PR DESCRIPTION
**Pull request by ashercodeorg**

### CODE REVIEW

**Comment by ashercodeorg** [comment]
Hmm. Hound is failing. But I don't know why or where. How do I find out?

File: dashboard/app/controllers/admin_reports_controller.rb [hunk 1]
```diff
@@ -17,21 +17,63 @@ def funometer
...
+    @script_ratings = all_ratings.joins("INNER JOIN scripts ON scripts.id = puzzle_ratings.script_id").group(:script_id).order('SUM(100.0 * rating) / COUNT(rating)').select('script_id', 'name', 'SUM(100.0 * rating) / COUNT(rating) AS percentage', 'COUNT(rating) AS cnt')
+
+    # Compute funometer percentages by level, saving the most-favored and
+    # least-favored with over one hundred ratings.
+    @level_headers = ['Script ID', 'Level ID', 'Script Name', 'Level Name', 'Percentage', 'Count']
+    level_ratings = all_ratings.joins("INNER JOIN scripts ON scripts.id = puzzle_ratings.script_id").joins("INNER JOIN levels ON levels.id = puzzle_ratings.level_id").group(:script_id, :level_id).select(:script_id, :level_id, 'scripts.name AS script_name', 'levels.name AS level_name', 'SUM(100.0 * rating) / COUNT(rating) AS percentage', 'COUNT(rating) AS cnt').having('cnt > ?', 100)
+    @favorite_level_ratings = level_ratings.order('SUM(100.0 * rating) / COUNT(rating) desc').limit(25)
+    @hated_level_ratings = level_ratings.order('SUM(100.0 * rating) / COUNT(rating) asc').limit(25)
+
+    render locals: {percentages_by_day: percentages_by_day.to_a.map{|k,v|[k.to_s,v.to_f]}}
```
Comment (Hamms) [inline dashboard/app/controllers/admin_reports_controller.rb:36]: given that you're using this same `map` pattern three times now, it might be worth pulling it out into a helper

File: dashboard/app/views/admin_reports/funometer_by_script.html.haml [hunk 2]
```diff
@@ -0,0 +1,68 @@
+%script{src:'https://www.google.com/jsapi'}
```
Comment (Hamms) [inline dashboard/app/views/admin_reports/funometer_by_script.html.haml]: It would probably be good to wrap this in a `- content_for :head do`

Comment (Hamms) [inline dashboard/app/views/admin_reports/funometer_by_script.html.haml]: On this view as well as the others

**Comment by Hamms** [comment]
Couple of very minor nits, otherwise LGTM!

File: dashboard/app/views/admin_reports/funometer_by_script.html.haml (hunk 2 above)
Comment (ashercodeorg) [inline dashboard/app/views/admin_reports/funometer_by_script.html.haml]: Done.

File: dashboard/app/controllers/admin_reports_controller.rb (hunk 1 above)
Comment (ashercodeorg) [inline dashboard/app/controllers/admin_reports_controller.rb:36]: Definitely, this whole stuff needs refactoring. Going to leave it for another CL, after discussing how best to attack a larger cleanup.
